Design.md

```diff
@@ -303,22 +303,60 @@ Get S3 capabilities of the platform
 
 ### getInformation
 Get the information of the REST Services, including platform name, OSIS version and etc (Static Details)
-1. First Version, Make all information configuration except the usage part
-1. Next Gen, Implement the usage tenant level part with UTAPIv2 (write -java utapi-client)
 
 ### updateOsisCaps (?)
 
 ### getBucketList
 Get the bucket list of the platform tenant
 * S3:`listBucket` API
-### getOsisUsage 
-Get the platform usage of global (without query parameter), tenant (with tenant_id) or user (only with user_id).
-1. We don’t know yet what needs to be done here
-1. For next generation,
-    1. Utapi Service level -> global platform
-    1. Utapi Account level -> tenant level
-    1. Utapi User level -> user level
 
+### getOsisUsage
+Get the platform usage of global level (without query parameter), tenant level (only with tenant_id) or user  level (with tenant_id and user_id).
+
+Platform usage has the 5 metrics: bucket count, object count, total_bytes, available_bytes, used_bytes at the level of global, tenant and user respectively.
+
+- bucket_count 
+  - global: get all accounts from VAULT which are associated with VCD tenants, 
+  and call OSIS getBucketList API of each tenant respectively to get the bucket count 
+  - tenant: call OSIS getBucketList API of the provided tenant_id to get the bucket count 
+  - user: call OSIS getBucketList API of the provided tenant_id to get all buckets then filter by bucket owner equal to provided user_id
+
+- object_count 
+  - global: get all accounts from VAULT which are associated with VCD tenants, 
+  call UTAPI ListMetrics API with all tenant_ids to get `numberOfObjects` field. 
+  (Split into multiple requests if the number of accounts is big, the threshold value is to be defined) 
+  - tenant: call UTAPI ListMetrics API with tenant_id to get `numberOfObjects` field 
+  - user: call UTAPI ListMetrics API with user_id to get `numberOfObjects` field
+
+- total_bytes 
+  - tenant: call VAULT getAccount API to get quota of the tenant, total_bytes will be the quota of the current tenant if quota exists, otherwise, it will be 0 
+  - global and user: 0
+
+- used_bytes 
+  - global: call UTAPI ListMetrics with all tenant_ids to get `storageUtilized` filed. (Split into multiple requests if the number of accounts is big, the threshold value is to be defined)
```

Comment:
@tmacro any recommendations on threshold?

Reply:
I'd start with 50-100 and see how that performs.

Reply:
fixed https://github.com/scality/osis/pull/115/commits/58e6a76186b8ed7e15c93d1801fb56c35c0a2abe